app/services/recent-events.ts

```diff
@@ -298,6 +404,15 @@ export class RecentEventsService extends StatefulService<IRecentEventsState> {
     this.state.recentEvents = events.concat(this.state.recentEvents);
   }
 
+  @mutation()
+  private TOGGLE_RECENT_EVENT_READ(eventHash: string) {
+    const matches = this.state.recentEvents.forEach((event, index) => {
+      if (event.hash === eventHash) {
+        this.state.recentEvents[index].read = !this.state.recentEvents[index].read;
+      }
+    });
```

Comment:
`matches` is not used.

Also, no need to modify with index like this.  You can mutate the object from within the `forEach`.

```TypeScript
    this.state.recentEvents.forEach(event => {
      if (event.hash === eventHash) {
        event.read = !event.read
      }
    });
```

Reply:
To elaborate, this works because objects are always passed by reference in javascript.

Reply:
Good point. Was using find earlier.